x/wasm/keeper/contract_keeper_test.go

```diff
@@ -0,0 +1,148 @@
+package keeper
+
+import (
+	"encoding/json"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/CosmWasm/wasmd/x/wasm/types"
+)
+
+func TestInstantiate2(t *testing.T) {
+	parentCtx, keepers := CreateTestInput(t, false, AvailableCapabilities)
+	example := StoreHackatomExampleContract(t, parentCtx, keepers)
+	otherExample := StoreReflectContract(t, parentCtx, keepers)
+
+	verifierAddr := RandomAccountAddress(t)
+	beneficiaryAddr := RandomAccountAddress(t)
+	initMsg := mustMarshal(t, HackatomExampleInitMsg{Verifier: verifierAddr, Beneficiary: beneficiaryAddr})
+
+	otherAddr := keepers.Faucet.NewFundedRandomAccount(parentCtx, sdk.NewInt64Coin("denom", 1_000_000_000))
+
+	// create instances for duplicate checks
+	exampleContract := func(t *testing.T, ctx sdk.Context, fixMsg bool) {
+		_, _, err := keepers.ContractKeeper.Instantiate2(
+			ctx,
+			example.CodeID,
+			example.CreatorAddr,
+			nil,
+			initMsg,
+			"my label",
+			sdk.NewCoins(sdk.NewInt64Coin("denom", 1)),
+			[]byte(`my salt`),
```

Comment:
nit, maybe make this a variable as you use it many places?

`mySalt := []byte("my salt")`

(totally not needed for merge, just a thought)
same with `label := "my label"`